### PR TITLE
added category crud

### PR DESCRIPTION
**Pull request by MuhammadSaqibAbro**

### CODE REVIEW

File: src/ui/ProductCategory.java [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+import model.ProductCategoryModel;
+
+
+/**
+ *
+ * @author SAQIB
+ */
+public class ProductCategory extends javax.swing.JFrame {
+    ProductCategoryDAO pcDAO=new ProductCategoryDAOImp();
+    Object columns[]={"IDE","Name","Code","Date","Created_by","Updated_by","Status"};
```
Comment (govindkmk) [inline src/ui/ProductCategory.java]: Make it Id

File: src/dao/ProductCategoryDao.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    
+    
+    
+    
+    
+    
+    
+    
+    
+
```
Comment (RaheemBux) [inline src/dao/ProductCategoryDao.java]: please remove extra empty lines from file and also do not write public keyword before method as interface methods are by default public

File: src/daoimp/ProductCategoryDAOImp.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+                       
+			e.printStackTrace();
+     }
+
+    
+}
+}
```
Comment (RaheemBux) [inline src/daoimp/ProductCategoryDAOImp.java]: please remove extra spaces from file or reformat the code.

File: src/model/ProductCategoryModel.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ */ 
+public class ProductCategoryModel {
+    private String category_name;
+    private String category_code;
+    private String created_date;
+    private Integer id;
+    private boolean active;
+    private String created_by;
+    private String updated_by;
+
```
Comment (RaheemBux) [inline src/model/ProductCategoryModel.java:21]: do not write Model word in the class name and please follow camelCase notation for variable names and also 
connect with @TouqeerAli for setting these fields
  private String created_date;
    private Integer id;
    private boolean active;
    private String created_by;
    private String updated_by;

**Review by TouqeerAli — APPROVED** [review]

**Review by Ahmadqazi584 — APPROVED** [review]

**Review by MuhammadHaseebOdho — APPROVED** [review]

**Pull request merged**